packages/db/tests/models/test-collection-model.js

```diff
@@ -79,3 +82,144 @@ test('CollectionPgModel.upsert() overwrites a collection record', async (t) => {
+  const records = times(3, (i) => fakeCollectionRecordFactory({
+    name: collectionName,
+    version: i,
+  }));
+
+  const dateValue = 5000;
+  const searchRecord = fakeCollectionRecordFactory({
+    name: collectionName,
+    version: '4',
+    updated_at: new Date(dateValue),
+  });
+  records.push(searchRecord);
+
+  await Promise.all(records.map((r) => collectionPgModel.create(knex, r)));
+
+  const searchResponse = await collectionPgModel.searchWithUpdatedAtRange(
+    knex,
+    {
+      name: collectionName,
+    },
+    {
+      updatedAtFrom: new Date(dateValue - 1),
+      updatedAtTo: new Date(dateValue + 1),
+    }
+  );
+
+  t.is(searchResponse.length, 1);
+  t.like(
+    removeNilProperties(searchResponse[0]),
+    { ...searchRecord, ...{ files: JSON.parse(searchRecord.files) } }
+  );
+});
+
+test('Collection.searchWithUpdatedAtRange() returns a filtered array of records if only updatedAtTo is specified', async (t) => {
+  const {
+    knex,
+    collectionPgModel,
+  } = t.context;
+
+  const dateValue = 5000;
+  const collectionName = cryptoRandomString({ length: 10 });
+  const records = times(3, (i) => fakeCollectionRecordFactory({
+    name: collectionName,
+    version: i,
+  }));
+
+  const searchRecord = fakeCollectionRecordFactory({
+    name: collectionName,
+    updated_at: new Date(dateValue),
+    version: '4',
+  });
+  records.push(searchRecord);
+
+  await Promise.all(records.map((r) => collectionPgModel.create(knex, r)));
+
+  const searchResponse = await collectionPgModel.searchWithUpdatedAtRange(
+    knex,
+    {
+      name: collectionName,
+    },
+    {
+      updatedAtTo: new Date(dateValue + 1),
+    }
+  );
+
+  t.is(searchResponse.length, 1);
+  t.like(
+    removeNilProperties(searchResponse[0]),
+    { ...searchRecord, ...{ files: JSON.parse(searchRecord.files) } }
+  );
+});
+
+test.serial('Collection.searchWithUpdatedAtRange() returns a filtered array of records if only updatedAtFrom is specified', async (t) => {
+  const {
+    knex,
+    collectionPgModel,
+  } = t.context;
+
+  const nowDateValue = new Date().valueOf();
+  const collectionName = cryptoRandomString({ length: 10 });
+  const records = times(3, (i) => fakeCollectionRecordFactory({
+    name: collectionName,
+    version: i,
+    updated_at: new Date(nowDateValue - 10000),
+  }));
+
+  const searchRecord = fakeCollectionRecordFactory({
+    updated_at: new Date(nowDateValue),
+    name: collectionName,
+    version: '4',
+  });
+  records.push(searchRecord);
+
+  await Promise.all(records.map((r) => collectionPgModel.create(knex, r)));
+
+  const searchResponse = await collectionPgModel.searchWithUpdatedAtRange(
+    knex,
+    {
+      name: collectionName,
+    },
+    {
+      updatedAtFrom: new Date(nowDateValue - 1),
+    }
+  );
+
+  t.is(searchResponse.length, 1);
+  t.like(
+    removeNilProperties(searchResponse[0]),
+    { ...searchRecord, ...{ files: JSON.parse(searchRecord.files) } }
```

Comment:
```suggestion
    { ...searchRecord, files: JSON.parse(searchRecord.files) }
```

Reply:
heh, good catch.     I don't rightly know how that came to be.